src/migration/1714633817175-SquadTotalPostsTrigger.ts

```diff
@@ -0,0 +1,67 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SquadTotalPostsTrigger1714633817175 implements MigrationInterface {
+  name = 'SquadTotalPostsTrigger1714633817175';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    queryRunner.query(`
+        CREATE OR REPLACE FUNCTION increment_squad_posts_count()
+          RETURNS TRIGGER
+          LANGUAGE PLPGSQL
+          AS
+        $$
+        DECLARE
+          s public.source;
+        BEGIN
+          SELECT * INTO s FROM public.source WHERE id = NEW."sourceId";
```

Comment:
I think better to just do `AND s.type = 'squad'` in UPDATE statement, that will avoid select query in each trigger execution if I see correctly?

Reply:
UPDATE that does not modify any rows should be faster then SELECT + OPTIONALLY UPDATE.

Reply:
Actually we can also use `WHEN (NEW.type = 'squad'` in trigger definition, that should be the best.

Reply:
But the type here is the post, so it will never be a `squad`.

Reply:
Yeah but you are doing UPDATE query on `source` where you can do a `WHERE` filter on its type.

Reply:
Oh, right. Forgot about that. Yes, we can do that.